src/test/java/itest/DiscoveryPluginIT.java

```diff
@@ -0,0 +1,204 @@
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import itest.bases.StandardSelfTest;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(OrderAnnotation.class)
+class DiscoveryPluginIT extends StandardSelfTest {
+
+    private static final Gson gson =
+            new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
+
+    static final NodeType NODE_TYPE =
+            new NodeType() {
+                @Override
+                public String getKind() {
+                    return "JVM";
+                }
+
+                @Override
+                public int ordinal() {
+                    return 0;
+                }
+            };
+
+    final String realm = getClass().getSimpleName();
+    final URI callback = URI.create("http://localhost:8181/");
+    private static volatile String id;
+
+    @Test
+    @Order(1)
+    void shouldBeAbleToRegister() throws InterruptedException, ExecutionException {
+        JsonObject body = new JsonObject(Map.of("realm", realm, "callback", callback));
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        webClient
+                .post("/api/v2.2/discovery")
+                .sendJson(
+                        body,
+                        ar -> {
+                            assertRequestStatus(ar, response);
+                            response.complete(ar.result().bodyAsJsonObject());
+                        });
+        JsonObject resp = response.get();
+        JsonObject info = resp.getJsonObject("data").getJsonObject("result");
+        DiscoveryPluginIT.id = info.getString("id");
+        MatcherAssert.assertThat(id, Matchers.notNullValue());
+    }
+
+    @Test
+    @Order(2)
+    void shouldBeAbleToUpdate() throws InterruptedException, ExecutionException {
+        JsonObject service = new JsonObject(Map.of("connectUrl", callback, "alias", "mynode"));
+        JsonObject target =
+                new JsonObject(
+                        Map.of(
+                                "target",
+                                service,
+                                "name",
+                                getClass().getSimpleName(),
+                                "nodeType",
+                                "JVM"));
+        JsonArray subtree = new JsonArray(List.of(target));
+
+        CompletableFuture<Buffer> response = new CompletableFuture<>();
+        webClient
+                .post("/api/v2.2/discovery/" + id)
+                .sendJson(
+                        subtree,
+                        ar -> {
+                            assertRequestStatus(ar, response);
+                            response.complete(ar.result().body());
+                        });
+        response.get();
```

Comment:
Should we check response status here too?

Reply:
It's hidden, but it does:

https://github.com/cryostatio/cryostat/blob/5517451aab07988b521887253b94364f34c478fe/src/test/java/itest/bases/StandardSelfTest.java#L126

This checks for `2xx` codes.